### ISSUE TITLE
Add appropriate spring.application.name to all generated apps

### ISSUE THREAD

**Issue by markpollack** [spring-cloud/spring-cloud-stream-app-starters#125]
This will enable configuration via Config Server.

**Comment by mbogoevici** [comment]
This would improve things, but not having it is not a deal breaker, IMO - we need to figure out how this would be provided by the dataflow server, anyway.

**Comment by markpollack** [comment]
If it isn't there, how would the application find the configuration in the config server.  The base case, would be to bootstrap the lookup of config data off spring.app.name that is baked into the .jar imho.  changing that at runtime via deployment props is nice, but not the base case I feel.

**Comment by mbogoevici** [comment]
If it's not there, it would only use the global `application.properties`

**Comment by markpollack** [comment]
and the global application.properties from what specifically?  If it is global, then the name will be the same for all the apps, which is bad.

**Comment by mbogoevici** [comment]
In the context of the config server, I meant this: http://cloud.spring.io/spring-cloud-config/spring-cloud-config.html - see "Sharing Configuration With All Applications" (no anchor to it) - if `spring.application.name` is absent (the default) - the application will still try to get the shared configuration - there's an immediate use for that in the form of global settings for binders, for example.

That being said, I don't mean that we shouldn't provide a `spring.application.name` in our apps - which would be equivalent to the module configuration from Spring XD. It's just that the absence of it doesn't render the config server support useless.

**Comment by markpollack** [comment]
I see.  The shared config is rather broad generally speaking.  Consider this use case...

```
s1 = http | jdbc (database1)
s2 = http | mytransformer | jdbc (database 2)
```

where database1 and database2 are different databases.  Even if we set the std OOTB app name to be jdbc-sink-kafka, it doesn't help us differentiate the config.  I guess we would need to pass in app.jdbc.spring.application.name=jdbc-sink-kafka-2 for the second one?

**Comment by mbogoevici** [comment]
Yeah, or maybe we can just use a different profile - e.g. `spring.profiles.include` ? which would override the default setting.

**Comment by markpollack** [comment]
Yes.  Profile would be better.

**Comment by mbogoevici** [comment]
The reason why I've suggested that we set them via SCDF is that in general we'd expect the user to know the label of the module, not its artifact name, e.g. `http` vs. `http-source`. The generation process can only set the latter.

**Comment by ericbottard** [comment]
>  I guess we would need to pass in app.jdbc.spring.application.name=jdbc-sink-kafka-2 for the second one when deploying?

The app name should be `<streamname>.<labelname>` don't you think? cc @markfisher
